ranger-gcs-permission-check-service/src/main/java/com/google/cloud/hadoop/ranger/gcs/permissionCheckService/ticketHandler/RangerHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *  * Copyright 2020 Google Inc. All Rights Reserved.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *    http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.google.cloud.hadoop.ranger.gcs.permissionCheckService.ticketHandler;
+
+import com.google.cloud.hadoop.ranger.gcs.permissionCheckService.RangerGcsPluginSingletonWrapper;
+import com.google.cloud.hadoop.ranger.gcs.permissionCheckService.utilities.RequestTicket;
+import com.google.cloud.hadoop.ranger.gcs.utilities.RangerGcsPermissionCheckResult;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Delegate authorization job to Ranger plugin.
+ */
+public class RangerHandler implements TicketHandler {
+    private static final Log LOG = LogFactory.getLog(RangerHandler.class);
+
+    @Override
+    public void handle(RequestTicket ticket) {
+        if (LOG.isDebugEnabled())
+            LOG.debug("==> RangerHandler.handle(" + ticket.toString() + ")");
+
+        if (ticket.getResult() != null && ticket.getResult().equals(RangerGcsPermissionCheckResult.Deny())) {
+            if (LOG.isDebugEnabled())
+                LOG.debug("<== RangerHandler.handle(" + ticket.toString() + "): Already denied by previous handler.");
```

Comment:
nit: I think "previous" is a bit confusing here. Can you re-phrase this error message a little so it is clearer request is denied by a handler in the chain?